### PR TITLE
Update offense analysis component

### PR DESCRIPTION
**Pull request by crc8109**

### CODE REVIEW

**Comment by crc8109** [comment]
@ay-a-git @jovalle if either of you could take a look, I'd appreciate it.

The point of the CLI app is that you can fig out what advantages/disadvantages your opponent has, based on the types of the pokemon currently out (i.e. one at a time)

File: offense_calculator.py [hunk 1]
```diff
@@ -1,734 +1,128 @@
...
-            "rock": 1,
-            "ghost": 1,
-            "dragon": 0.5,
-            "dark": 1,
-            "steel": 1,
-            "fairy": 1
+            2: ["water", "flying"],
+            1: ["normal", "fire", "ice", "fighting", "poison", "psychic", "bug", "rock", "dark", "steel", "fairy", "none"],
+            0.5: ["electric", "grass", "dragon", "ground"],
+            0: ["ground"]
```
Comment (ay-a-git) [inline offense_calculator.py:26]: add Rock type towards the 0 value. Elec doesnt affect rock or ground

File: offense_calculator.py [hunk 2]
```diff
@@ -1,734 +1,128 @@
...
-            "rock": 0.5,
-            "ghost": 1,
-            "dragon": 1,
-            "dark": 1,
-            "steel": 0.5,
-            "fairy": 1
+            2: ["grass", "fighting", "bug"],
+            1: ["normal", "fire", "water", "ice", "poison", "ground", "flying", "psychic", "ghost", "dragon", "dark", "fairy", "none"],
+            0.5: ["electric", "rock", "steel"],
+            0: ["ground"]
```
Comment (ay-a-git) [inline offense_calculator.py:60]: i believe flying type moves do half damage on ground types

File: offense_calculator.py (hunk 1 above)
Comment (crc8109) [reply offense_calculator.py:26]: https://pokemondb.net/type

File: offense_calculator.py (hunk 2 above)
Comment (crc8109) [reply offense_calculator.py:60]: https://pokemondb.net/type

**Pull request merged**